sw/host/tests/chip/flash_ctrl/src/main.rs

```diff
@@ -0,0 +1,293 @@
+    LcCtrlTransitionRegwen,
+};
+
+use std::collections::HashMap;
+use std::fs;
+use std::path::PathBuf;
+use std::time::Duration;
+
+use object::{Object, ObjectSymbol};
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::execute_test;
+use opentitanlib::test_utils::init::InitializeTest;
+use opentitanlib::test_utils::mem::{MemRead32Req, MemWrite32Req};
+use opentitanlib::uart::console::UartConsole;
+use opentitanlib::io::jtag::JtagTap;
+use opentitanlib::test_utils::load_sram_program::{ExecutionMode, SramProgramParams
+};
+
+#[derive(Debug, Parser)]
+struct Opts {
+    #[command(flatten)]
+    init: InitializeTest,
+
+    /// Console receive timeout.
+    #[arg(long, value_parser = humantime::parse_duration, default_value = "600s")]
+    timeout: Duration,
+
+    /// Path to the firmware's ELF file, for querying symbol addresses.
+    #[arg(value_name = "FIRMWARE_ELF")]
+    firmware_elf: PathBuf,
+
+    #[command(flatten)]
+    sram_program: SramProgramParams,
+
+}
+
+fn test_update_phase(
+    _opts: &Opts,
+    test_word_address: u32,
+    transport: &TransportWrapper,
+    value: u32,
+) -> Result<()> {
+    let uart = transport.uart("console")?;
+    uart.set_flow_control(true)?;
+    let _ = UartConsole::wait_for(&*uart, r"Starting [^\r\n]*", _opts.timeout)?;
+    let _ = uart.clear_rx_buffer();
+    MemWrite32Req::execute(&*uart, test_word_address, value)?;
+    Ok(())
+}
+
+fn test_end(opts: &Opts, end_test_address: u32, transport: &TransportWrapper) -> Result<()> {
+    let uart = transport.uart("console")?;
+    let end_test_value = MemRead32Req::execute(&*uart, end_test_address)?;
+    assert!(end_test_value == 0);
+    MemWrite32Req::execute(&*uart, end_test_address, /*value=*/ 1)?;
+    let _ = UartConsole::wait_for(&*uart, r"test_end[^\r\n]*", opts.timeout)?;
+    Ok(())
+}
+
+fn test_sram_load(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
+    //
+    // Connect to the RISC-V TAP
+    //
+    transport.pin_strapping("PINMUX_TAP_RISCV")?.apply()?;
+    transport.reset_target(opts.init.bootstrap.options.reset_delay, true)?;
+
+    log::info!("Connecting to RISC-V TAP");
+    let mut jtag = opts
+        .init
+        .jtag_params
+        .create(transport)?
+        .connect(JtagTap::RiscvTap)?;
+
+    log::info!("Halting core");
+    jtag.halt()?;
+
+    opts.sram_program
+        .load_and_execute(&mut *jtag, ExecutionMode::Jump)?;
+
+    jtag.disconnect()?;
+    Ok(())
+}
+
+fn test_rma_command(opts: &Opts, transport: &TransportWrapper) -> anyhow::Result<()> {
+    let rma_bootstrap_strapping = transport.pin_strapping("RMA_BOOTSTRAP")?;
+    let tap_strapping = transport.pin_strapping("PINMUX_TAP_LC")?;
+
+    // Need to reset with `RMA_BOOTSTRAP` set.
+    rma_bootstrap_strapping
+        .apply()
+        .context("failed to apply RMA_BOOTSTRAP strapping")?;
+
+    // We keep `PINMUX_TAP_LC` set so that resets due to transition don't
```

Comment:
I am a bit confused by this comment: I understand that you need to set the TAP straps before the reset since you can't change it dynamically in DEV mode. But when you reset the chip, the LC tap *will* be disconnected for a short time.

Reply:
It was copied from [this](https://github.com/lowRISC/opentitan/blob/927176403871e9504681b0dd080ae8c525411428/sw/host/tests/rom/e2e_bootstrap_rma/src/main.rs#L122)